cpp/dolfinx/fem/interpolate.h

```diff
@@ -110,32 +104,97 @@ void interpolate_from_any(Function<T>& u, const Function<T>& v)
     throw std::runtime_error(
         "Interpolation on different meshes not supported (yet).");
   }
+
   const int tdim = mesh->topology().dim();
 
-  // Get dofmaps
-  assert(v.function_space());
-  std::shared_ptr<const fem::DofMap> dofmap_v = v.function_space()->dofmap();
-  assert(dofmap_v);
+  const std::shared_ptr<const FiniteElement> element_to
+      = u.function_space()->element();
+  const std::shared_ptr<const FiniteElement> element_from
+      = v.function_space()->element();
+
   auto map = mesh->topology().index_map(tdim);
   assert(map);
 
-  std::vector<T>& coeffs = u.x()->mutable_array();
-
-  // Iterate over mesh and interpolate on each cell
-  const auto dofmap_u = u.function_space()->dofmap();
-  const std::vector<T>& v_array = v.x()->array();
-  const int num_cells = map->size_local() + map->num_ghosts();
-  const int bs = dofmap_v->bs();
-  assert(bs == dofmap_u->bs());
-  for (int c = 0; c < num_cells; ++c)
+  if (element_to->hash() == element_from->hash())
```

Comment:
This should be simplified to a copy of array, instead of looping over all dofs, as hash0==has1 means same function space?

Reply:
the data can be strided for sub-elements.